lib/common/ipc_server.c

```diff
@@ -316,6 +316,11 @@ pcmk__free_client(pcmk__client_t *c)
         if (c->remote->auth_timeout) {
             g_source_remove(c->remote->auth_timeout);
         }
+#ifdef HAVE_GNUTLS_GNUTLS_H
+        if (c->remote->tls_session != NULL) {
+            gnutls_free(c->remote->tls_session);
+        }
+#endif  // HAVE_GNUTLS_GNUTLS_H
```

Comment:
Does "this change" mean what's in the PR or what you've suggested?

It is odd for `pcmk__free_client()` not to free everything dynamically allocated within the client object, so I feel like that part should stay. Adjustments may need to be made elsewhere, since that hasn't been done until now.

Reply:
What's in the PR. Yeah tls_session needs to be NULLed wherever it's freed

Reply:
Not sure if NULL is the only issue, but it's the most obvious

Reply:
I have Pacemaker running under valgrind on both cluster nodes and it's not reporting anything, which is odd if double-free is the main issue. Agree it should still be NULLed, I'll see what happens afterward.

Reply:
never mind, I thought I was safe to skip valgrind on the remote node but the issue is in remoted_proxy

Reply:
There was only one place where it wasn't already being NULLed, and that appears to fix the issue.